[IMP] mrp: overhaul bom basics 17.0 (#9239)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/basic_setup/bill_configuration.rst

.. _manufacturing/basic_setup/setup-operations:

Operations
----------

Add an :guilabel:`Operation` to a |BOM| to specify instructions for production and register time
spent on an operation. To use this feature, first enable the *Work Orders* feature by going to
:menuselection:`Manufacturing app --> Configuration --> Settings`. In the :guilabel:`Operations`
section, tick the :guilabel:`Work Orders` checkbox to enable the feature.

.. seealso::
   :doc:`../advanced_configuration/work_order_dependencies`

.. image:: bill_configuration/enable-work-orders.png
   :align: center
   :alt: "Work Orders" feature in the Settings page.

Next, navigate to the |BOM| by going to :menuselection:`Manufacturing app --> Products --> Bill of
Materials` and selecting the desired |BOM|. To add a new operation, go to the :guilabel:`Operations`
tab, and click :guilabel:`Add a line`.

Doing so opens the :guilabel:`Create Operations` pop-up window, where the various fields of the
operation are configured:

- :guilabel:`Operation`: name of the operation.
- :guilabel:`Work Center`: select existing locations to perform the operation, or create a new work

>   center by typing the name and selecting the :guilabel:`Create...` option.

Author: ```suggestion
  center by typing the name and selecting the :guilabel:`Create " "` option.
```

- :guilabel:`Apply on Variants`: specify if this operation is only available for certain product
  variants. If the operation applies to all product variants, leave this field blank.

.. seealso::
   :doc:`Configuring BoMs for product variants <../advanced_configuration/product_variants>`

- :guilabel:`Duration Computation`: choose how time spent on the operation is tracked. Opt for
  :guilabel:`Compute based on tracked time` use the operation's time tracker or :guilabel:`Set

>   duration manually` if operators can record and modify time themselves.

Author: ```suggestion
  duration manually` if operators can record and modify time themselves. TODO: Choosing the Compute based on tracked time option enables the Based on last `10` work orders option, which automatically estimates the time to complete this operation based on the last few operations. Choosing Set duration manually enables the Default Duration field instead.
```

- :guilabel:`Default Duration`: estimated amount of time to complete the operation; used for
  `planning manufacturing orders <https://www.youtube.com/watch?v=TK55jIq00pc>`_ and determining
  `work center availability <https://www.youtube.com/watch?v=3YwFlD97Bio>`_.
- :guilabel:`Company`: specify the company the |BOM| is available in.

Include operation details in the :guilabel:`Work Sheet` tab. Choose :guilabel:`PDF` to attach a file
or :guilabel:`Google Slide` to share a link. Select :guilabel:`Text` to type instructions in the

> :guilabel:`Description` text field.

Author: ```suggestion
:guilabel:`Description` text field. TODO: Type `/` for a list of formatting options and features, including ChatGPT.
```

.. image:: bill_configuration/create-operations.png
   :align: center
   :alt: Fill out the Create Operations pop-up window.

> Finally, click :guilabel:`Save & Close` to close the pop-up window. To add more operations, click

Reviewer: @StraubCreative and @Felicious it is not necessary to escape ampersands in RST (unless it is in an HTML context like using the :raw: role/directive). In cases where it is used in paragraphs and plain text roles like :guilabel:s, it does not need to be escaped since it is treated as just a plain text special character (just like $, %, #, etc), and it is not considered an [inline markup character](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup) either.

Example on the context:
<img width="232" alt="Screenshot 2024-05-17 at 11 02 04 AM" src="https://github.com/odoo/documentation/assets/36018073/cb3587ca-ee02-4467-9893-fbb1d9c1c211">

This is the output when rendered:
<img width="202" alt="Screenshot 2024-05-17 at 11 01 39 AM" src="https://github.com/odoo/documentation/assets/36018073/7b2e0a7c-765f-4626-8e4e-560582dd077a">

You can see the unicode character is only rendered in the HTML context, since the ampersand in that context indicates the beginning of an entity reference or numeric character reference. 

All that to say, escaping ampersands in plain text doesn't hurt– its just not necessary!

:guilabel:`Save & New` and repeat the same steps above to configure another operation.

.. note::
   Each operation is unique, as it is always exclusively linked to one |BOM|.

.. tip::
   After creating an operation, click the :guilabel:`Copy Existing Operations` button to choose an
   operation to duplicate.

   .. image:: bill_configuration/copy-existing-operations.png
      :align: center
      :alt: Show Operation tab, highlighting the "Copy Existing Operations" field.

Instructions
~~~~~~~~~~~~

.. important::
   To add detailed instructions to operations, the *Quality* app must be installed.

Add specific instructions to an existing operation by clicking the operation's |list| icon in the
:guilabel:`Instructions` column. The number in the :guilabel:`Instructions` column shows the number
of existing detailed instructions there are for the operation.

.. image:: bill_configuration/add-instructions.png
   :align: center
   :alt: Show the Instructions column, and list icon.

> Click :guilabel:`New` to open a quality control point form. Here, give the specific instruction a
> :guilabel:`Title` and change the :guilabel:`Type` to :guilabel:`Instructions`. For more details
> about quality control points refer to the following:
> 
> .. seealso::
>    :doc:`Instruction check <../../quality/quality_check_types/instructions_check>`

Reviewer 2: Suggestions around specificity, depth, and formatting.

```suggestion
On the :guilabel:`Steps` dashboard, click :guilabel:`New` to open a blank quality control point form
where the new manufacturing step can be created. Here, give the specific instruction a
:guilabel:`Title` and set the :guilabel:`Type` to :guilabel:`Instructions`. In the
:guilabel:`Instructions` tab of the form, write out the directions for the step in the operation.

.. seealso:
   Further customizations can be made here on this form, beyond ordinary instructions, to also
   include specific types of quality control points that carry specific (or complex) conditions. For
   more details about quality control points refer to the following document: :doc:`Instruction
   check <../../quality/quality_check_types/instructions_check>`
```

.. image:: bill_configuration/steps.png
   :align: center
   :alt: Show the page to add a quality check.

> 

Author: ```suggestion
TODO:
Miscellaneous
---------------- 
```

Add by-products to BoMs
=======================

A *by-product* is a residual product that is created during production in addition to the main
product of a |BOM|. Unlike the primary product, there can be more than one by-product on a |BOM|.

To add by-products to a |BOM|, first enable the :guilabel:`By-Products` feature in
:menuselection:`Manufacturing app --> Configuration --> Settings`. In the :guilabel:`Operations`
section, tick the checkbox for :guilabel:`By-Products` to enable the feature.

.. image:: bill_configuration/by-products.png
   :align: center
   :alt: "By Products" feature in the settings page.

Once the feature is enabled, add by-products to a |BOM| by clicking the :guilabel:`By-products` tab.
Click :guilabel:`Add a line`, and fill in the :guilabel:`By-product`, :guilabel:`Quantity`, and
:guilabel:`Unit of Measure`. Optionally, specify a :guilabel:`Produced in Operation` for the
by-product.

.. example::
   The by-product, `Mush`, is created in the `Grind grapes` operation when producing `Red Wine`.

>    .. image:: bill_configuration/add-by-product.png

Reviewer 2: I love this example 🤣

Bill of materials
=================

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |list| replace:: :icon:`fa-list-ul` :guilabel:`(list)`

.. _manufacturing/management/bill-configuration:

> A *bill of materials* (or *BoM* for short) is a document component quantities needed for a product,

Reviewer 3: ```suggestion
A *bill of materials* (or *BoM* for short) documents the component quantities needed for a product,
```

option.

.. image:: bill_configuration/component.png
   :align: center
   :alt: Add a component by selecting it from the drop-down menu.

.. _manufacturing/basic_setup/setup-operations:

Operations
----------

Add an :guilabel:`Operation` to a |BOM| to specify instructions for production and register time
spent on an operation. To use this feature, first enable the *Work Orders* feature by going to
:menuselection:`Manufacturing app --> Configuration --> Settings`. In the :guilabel:`Operations`
section, tick the :guilabel:`Work Orders` checkbox to enable the feature.

.. seealso::
   :doc:`../advanced_configuration/work_order_dependencies`

.. image:: bill_configuration/enable-work-orders.png
   :align: center
   :alt: "Work Orders" feature in the Settings page.

Next, navigate to the |BOM| by going to :menuselection:`Manufacturing app --> Products --> Bill of
Materials` and selecting the desired |BOM|. To add a new operation, go to the :guilabel:`Operations`
tab, and click :guilabel:`Add a line`.

Doing so opens the :guilabel:`Create Operations` pop-up window, where the various fields of the

> operations are configured:

Reviewer 3: ```suggestion
operation are configured:
```

Include operation details in the :guilabel:`Work Sheet` tab. Choose :guilabel:`PDF` to attach a file
or :guilabel:`Google Slide` to share a link. Select :guilabel:`Text` to type instructions in the
:guilabel:`Description` text field.

.. image:: bill_configuration/create-operations.png
   :align: center
   :alt: Fill out the Create Operations pop-up window.

Finally, click :guilabel:`Save & Close` to close the pop-up window. To add more operations, click
:guilabel:`Save & New` and repeat the same steps above to configure another operation.

.. note::
   Each operation is unique, as it is always exclusively linked to one |BOM|.

.. tip::
   After creating an operation, click the :guilabel:`Copy Existing Operations` button to choose an
   operation to duplicate.

   .. image:: bill_configuration/copy-existing-operations.png
      :align: center
      :alt: Show Operation tab, highlighting the "Copy Existing Operations" field.

Instructions
~~~~~~~~~~~~

.. important::
   To add detailed instructions to operations, the *Quality* app must be installed.

Add specific instructions to an existing operation by clicking the operation's |list| icon in the

> :guilabel:`Instructions` column. The number in the :guilabel:`Instructions` column shows the amount

Reviewer 3: ```suggestion
:guilabel:`Instructions` column. The number in the :guilabel:`Instructions` column shows the number
```

- :guilabel:`Work Center`: select existing locations to perform the operation, or create a new work
  center by typing the name and selecting the :guilabel:`Create " "` option.
- :guilabel:`Apply on Variants`: specify if this operation is only available for certain product
  variants. If the operation applies to all product variants, leave this field blank.

.. seealso::
   :doc:`Configuring BoMs for product variants <../advanced_configuration/product_variants>`

- :guilabel:`Duration Computation`: choose how time spent on the operation is tracked. Opt for
  :guilabel:`Compute based on tracked time` use the operation's time tracker or :guilabel:`Set
  duration manually` if operators can record and modify time themselves.
- :guilabel:`Default Duration`: estimated amount of time to complete the operation; used for
  `planning manufacturing orders <https://www.youtube.com/watch?v=TK55jIq00pc>`_ and determining
  `work center availability <https://www.youtube.com/watch?v=3YwFlD97Bio>`_.
- :guilabel:`Company`: specify the company the |BOM| is available in.

Include operation details in the :guilabel:`Work Sheet` tab. Choose :guilabel:`PDF` to attach a file

> or :guilabel:`Google Slide` to share a link. Select :guilabel:`Text` to type instructions in the

Author: ```suggestion
or :guilabel:`Google Slide` with *public* access to share a link. Select :guilabel:`Text` to type instructions in the
```

   :align: center
   :alt: Fill out the Create Operations pop-up window.

Finally, click :guilabel:`Save & Close` to close the pop-up window. To add more operations, click
:guilabel:`Save & New` and repeat the same steps above to configure another operation.

.. note::
   Each operation is unique, as it is always exclusively linked to one |BOM|.

.. tip::
   After creating an operation, click the :guilabel:`Copy Existing Operations` button to choose an
   operation to duplicate.

   .. image:: bill_configuration/copy-existing-operations.png
      :align: center
      :alt: Show Operation tab, highlighting the "Copy Existing Operations" field.

Instructions
~~~~~~~~~~~~

.. important::
   To add detailed instructions to operations, the *Quality* app must be installed.

Add specific instructions to an existing operation by clicking the operation's |list| icon in the
:guilabel:`Instructions` column. The number in the :guilabel:`Instructions` column shows the number
of existing detailed instructions there are for the operation.

.. image:: bill_configuration/add-instructions.png
   :align: center
   :alt: Show the Instructions column, and list icon.

> Click :guilabel:`New` to open a quality control point form. Here, give the specific instruction a
> :guilabel:`Title` and change the :guilabel:`Type` to :guilabel:`Instructions`. For more details
> about quality control points refer to the following:

Author: ```suggestion
On the :guilabel:`Steps` dashboard, click :guilabel:`New` to open a blank quality control point form
where the new manufacturing step can be created. Here, give the specific instruction a
:guilabel:`Title` and set the :guilabel:`Type` to :guilabel:`Instructions`. In the
:guilabel:`Instructions` tab of the form, write out the directions for the step in the operation.
```

> .. seealso::
>    :doc:`Instruction check <../../quality/quality_check_types/instructions_check>`

Author: ```suggestion
.. note:
   Further customizations can be made here on this form, beyond ordinary instructions, to also
   include specific types of quality control points that carry specific (or complex) conditions. For
   more details about quality control points refer to the following document: :doc:`Instruction
   check <../../quality/quality_check_types/instructions_check>`
```

following features to enable these columns:

- :guilabel:`Apply on Variants`: specify which :doc:`product variant
  <../advanced_configuration/product_variants>` each component is used in. When the field is left
  blank, the component is used in all product variants.

.. _manufacturing/basic_setup/consumed-in-operation:

- :guilabel:`Consumed in Operation`: specify the operation using the component. Useful for
  determining :ref:`manufacturing readiness <manufacturing/basic_setup/manufacturing-readiness>`.
- :guilabel:`Manual Consumption`: tick the checkbox to force operators to check the
  :guilabel:`Consumed` checkbox on a manufacturing order (MO).

  .. image:: bill_configuration/consumed-field.png
     :align: center
     :alt: Show a manufacturing order, highlighting the *Consumed* field.

  Not doing so triggers the :guilabel:`Consumption Warning` error message, where the consumed
  component quantity must be manually inputted. Otherwise, the operation cannot be completed.

  .. image:: bill_configuration/consumption-warning.png
     :align: center
     :alt: Show the consumption warning error message.

.. _manufacturing/basic_setup/setup-operations:

Operations
----------

> Add an :guilabel:`Operation` to a |BOM| to specify instructions for production and register time
> spent on an operation. To use this feature, first enable the *Work Orders* feature by going to

Reviewer: Nitpick– the text "Operation" is not visible on the current UI and because in this context you are referring to the Operations feature, and not necessarily the Operations tab
```suggestion
Add an *Operation* to a |BOM| to specify instructions for production and register time spent on an
operation. To use this feature, first enable the *Work Orders* feature by going to
```

  available` shows the :guilabel:`Component Status` as a **green** :guilabel:`Not Available`, when
  only the components that are consumed in the first operation are in stock. This indicates that
  although not all components are available, operators can at least begin with the first operation.
  Choosing :guilabel:`When all components are available` displays a **red** :guilabel:`Not
  Available` component status unless all components are in available.

  .. tip::
     Specify which operation consumes each component on the |BoM| in the :ref:`Manual Consumption
     field <manufacturing/basic_setup/consumed-in-operation>`.

  .. image:: bill_configuration/component-status.png
     :align: center
     :alt: Show the *Component Status* field on the manufacturing order dashboard.

- :guilabel:`Version`: displays the current |BoM| version, visible with the Odoo *PLM* app installed
  for managing |BoM| changes.
- :guilabel:`Flexible Consumption`: specifies if components used can deviate from the quantity
  defined on the |BoM|. Choose :guilabel:`Blocked` if operators **must** adhere strictly to the

>   |BoM| quantity. otherwise, choose :guilabel:`Allowed` or :guilabel:`Allowed with Warning`.

Reviewer: Capitalize "Otherwise" :)
```suggestion
  |BoM| quantity. Otherwise, choose :guilabel:`Allowed` or :guilabel:`Allowed with Warning`.
```

- :guilabel:`Routing`: select the preferred warehouse's manufacturing operation type for products
  produced in multiple warehouses. If left blank, this warehouse's `Manufacturing` operation type is
  used by default.
- :guilabel:`Analytic Distribution`: select pre-created :doc:`analytic distribution models
  <../../../finance/accounting/reporting/analytic_accounting>` from the list to automatically record
  the cost of manufacturing products in the chosen journal.
- :guilabel:`Manuf Lead Time`: define the number of days needed to complete a |MO| from the date of
  confirmation.
- :guilabel:`Days to prepare Manufacturing Order`: number of days needed to replenish components, or
  manufacture sub-assemblies of the product.

.. seealso::
   - :doc:`Analytic distribution <../../../finance/accounting/reporting/analytic_accounting>`
   - :doc:`Lead times
     <../../inventory/shipping_receiving/advanced_operations_shipping/scheduled_dates>`

.. image:: bill_configuration/misc-tab.png
   :align: center
   :alt: Show the *Miscellaneous* tab of the BoM.

> 

Author: ```suggestion
```

Add by-products to BoMs
=======================

A *by-product* is a residual product that is created during production in addition to the main
product of a |BOM|. Unlike the primary product, there can be more than one by-product on a |BOM|.

> To add by-products to a |BOM|, first enable the :guilabel:`By-Products` feature in

Reviewer: Nitpick- the "By-Products" text is not visible yet on the current UI
```suggestion
To add by-products to a |BOM|, first enable the *By-Products* feature in
```

:menuselection:`Manufacturing app --> Configuration --> Settings`. In the :guilabel:`Operations`
section, tick the checkbox for :guilabel:`By-Products` to enable the feature.

.. image:: bill_configuration/by-products.png
   :align: center
   :alt: "By Products" feature in the settings page.

Once the feature is enabled, add by-products to a |BOM| by clicking the :guilabel:`By-products` tab.
Click :guilabel:`Add a line`, and fill in the :guilabel:`By-product`, :guilabel:`Quantity`, and
:guilabel:`Unit of Measure`. Optionally, specify a :guilabel:`Produced in Operation` for the
by-product.

.. example::

>    The by-product, `Mush`, is created in the `Grind grapes` operation when producing `Red Wine`.

Reviewer: I love this example! 😄

Fun fact– the _mush_ is called "Lees" https://www.wineenthusiast.com/basics/lees-wine-meaning/

Author: oooh I'm becoming more *cultured* :wink: just by reading about this!
